extensions/blocks/google-calendar/edit.js

```diff
@@ -0,0 +1,193 @@
+/**
+ * External dependencies
+ */
+import { __, _x } from '@wordpress/i18n';
+import { Component } from '@wordpress/element';
+import {
+	Placeholder,
+	SandBox,
+	Button,
+	Notice,
+	ExternalLink,
+	PanelBody,
+} from '@wordpress/components';
+import { BlockIcon, InspectorControls } from '@wordpress/block-editor';
+
+/**
+ * Internal dependencies
+ */
+import { icon } from '.';
+import { isMobile } from '../../../_inc/client/lib/viewport';
+import { extractAttributesFromIframe, IFRAME_REGEX, URL_REGEX } from './utils';
+
+class GoogleCalendarEdit extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.state = {
+			editedEmbed: this.props.attributes.url || '',
+			editingUrl: false,
+			interactive: false,
+			notice: null,
+		};
+	}
+
+	static getDerivedStateFromProps( nextProps, state ) {
```

Comment:
If we're requiring a more recent version of WP/Gutenberg then we could refactor this to a functional component with hooks, and we may find that it simplifies some of the internal state.

Reply:
@pento do you have any preferences for refactoring this to a functional component or not?

Reply:
If it's going to take time/effort to refactor, I don't think it's worth it. Ideally, this will be entirely deprecated by WordPress/gutenberg#19113 (example, #14648).

Reply:
>If it's going to take time/effort to refactor, I don't think it's worth it 

I tend to agree, lets leave it as is until WordPress/gutenberg#19113 is available.

Reply:
Agreed, it doesn't seem worth it. That's reminded me to have a look at  WordPress/gutenberg#19113